### PR TITLE
adding new integration

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- New integration: (╯°□°)╯︵ ┻━┻
# Screenshots / Videos

Screenshot 2022-11-16 at 17 05 59

### CODE REVIEW

File: src/Blocks/components/form/assets/form.js [hunk 1]
```diff
@@ -588,6 +591,7 @@ export class Form {
 		// Set error classes and error text on fields which have validation errors.
 		for (const [key] of Object.entries(fields)) {
 			const item = element.querySelector(`${this.errorSelector}[data-id="${key}"]`);
+			console.log(element);
```
Comment (mbmjertan) [inline src/Blocks/components/form/assets/form.js:594]: Stray console log?

```suggestion
```

File: src/Blocks/custom/airtable/components/airtable-options.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		settingsPageUrl,
+	} = select(STORE_NAME).getSettings();
+
+	const wpAdminUrl = esFormsLocalization.wpAdminUrl;
+
+	return (
+		<PanelBody title={__('Airtable', 'eightshift-forms')}>
+			<BaseControl
+				label={<IconLabel icon={icons.options} label={__('Settings', 'eightshift-forms')} />}
+				help={__('On Airtable settings page you can setup all details regarding you integration.', 'eightshift-forms')}
```
Comment (mbmjertan) [inline src/Blocks/custom/airtable/components/airtable-options.js:20]: On the Airtable settings page, you can set up all details regarding your integration.

File: src/Integrations/Airtable/SettingsAirtable.php [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+						'tabLabel' => \__('Help', 'eightshift-forms'),
+						'tabContent' => [
+							[
+								'component' => 'steps',
+								'stepsTitle' => \__('How to get the API key?', 'eightshift-forms'),
+								'stepsContent' => [
+									\__('Log in to your Airtable Account.', 'eightshift-forms'),
+									// translators: %s will be replaced with the link.
+									\sprintf(\__('Go to <a target="_blank" rel="noopener noreferrer" href="%s">Developers Hub</a>.', 'eightshift-forms'), 'https://airtable.com/create/tokens/new'),
+									\__('Create a new Personal access token with this scopes <strong>"data.records:write"</strong> and <strong>"schema.bases:read"</strong>.', 'eightshift-forms'),
```
Comment (mbmjertan) [inline src/Integrations/Airtable/SettingsAirtable.php:297]: the scopes* or these scopes* (but preferrably, the scopes)

File: src/Settings/SettingsHelper.php [hunk 4]
```diff
@@ -1064,4 +1064,53 @@ function ($option) use ($formId, $key) {
...
+		array $items,
+		string $selectedFormId,
+		string $key
+	): array {
+		return [
+			[
+				'component' => 'select',
+				'selectName' => $this->getSettingsName($key),
+				'selectId' => $this->getSettingsName($key),
+				'selectFieldLabel' => \__('Select sub form', 'eightshift-forms'),
```
Comment (mbmjertan) [inline src/Settings/SettingsHelper.php:1089]: I am not too sure what is a "sub form", but the name should probably be changed to something a bit more intuitive and appropriate.

File: src/Blocks/custom/airtable/manifest.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+	"blockName": "airtable",
+	"title": "Airtable Form",
+	"description" : "A Airtable subscription form",
```
Comment (mbmjertan) [inline src/Blocks/custom/airtable/manifest.json:4]:
```suggestion
	"description" : "An Airtable subscription form",
```

File: src/Integrations/Airtable/SettingsAirtable.php [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+						'component' => 'tab',
+						'tabLabel' => \__('Help', 'eightshift-forms'),
+						'tabContent' => [
+							[
+								'component' => 'steps',
+								'stepsTitle' => \__('How to get the API key?', 'eightshift-forms'),
+								'stepsContent' => [
+									\__('Log in to your Airtable Account.', 'eightshift-forms'),
+									// translators: %s will be replaced with the link.
+									\sprintf(\__('Go to <a target="_blank" rel="noopener noreferrer" href="%s">Developers Hub</a>.', 'eightshift-forms'), 'https://airtable.com/create/tokens/new'),
```
Comment (mbmjertan) [inline src/Integrations/Airtable/SettingsAirtable.php:296]:
```suggestion
									\sprintf(\__('Go to the <a target="_blank" rel="noopener noreferrer" href="%s">Developers Hub</a>.', 'eightshift-forms'), 'https://airtable.com/create/tokens/new'),
```

**Review by mbmjertan — COMMENTED** [review]
Looking good 💯 

Left a few language tips, nothing too major, and no complaints apart from that. Leaving as a comment not to block because of grammar.

File: src/Blocks/components/submit/submit.php [hunk 7]
```diff
@@ -60,7 +59,17 @@
...
-$submitIconContent = !empty($submitIcon) && $manifest['icons'][$submitIcon] ? $manifest['icons'][$submitIcon] : '';
+$submitIconContent = '';
+if (!empty($submitIcon)) {
+	$manifestIcon = $manifest['icons'][$submitIcon] ?? '';
+
+	if ($manifestIcon) {
+		$submitIconContent = $manifestIcon;
+	} else {
+		$submitIconContent = $submitIcon;
+	}
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/submit/submit.php:70]:
```suggestion
	$submitIconContent = $manifest['icons'][$submitIcon] ?? $submitIcon;
```

File: src/Blocks/custom/airtable/manifest.json (hunk 5 above)
Comment (goranalkovic-infinum) [reply src/Blocks/custom/airtable/manifest.json:4]: maybe just _Airtable subscription form_? 😄

**Review by goranalkovic-infinum — APPROVED** [review]
Just a couple of minor things, nice job!

**Pull request merged**